pyodide-build/pyodide_build/tests/test_config.py

```diff
@@ -62,9 +62,31 @@ def test_load_config_from_env(self):
         assert config["cmake_toolchain_file"] == "/path/to/toolchain"
         assert config["meson_cross_file"] == "/path/to/crossfile"
 
-    def test_load_config_from_file(self):
-        # TODO: Implement this
-        assert True
+    def test_load_config_from_file(self, tmp_path, reset_env_vars, reset_cache):
+        pyproject_file = tmp_path / "pyproject.toml"
+
+        env = {
+            "MESON_CROSS_FILE": "/path/to/crossfile",
+            "CFLAGS_BASE": "-O2",
+        }
+
+        pyproject_file.write_text("""[tool.pyodide.build]
+                                  invalid_flags = "this_should_not_be_parsed"
```

Comment:
Maybe it would be better to show some warning if an invalid flag is used. Otherwise, it will be hard to detect typo.